src/modules/Game.class.js

```diff
@@ -1,68 +1,265 @@
+
+        while (updatedColumn.length < 4) {
+          updatedColumn.push(0);
+        }
+
+        for (let r = 0; r < 4; r++) {
+          if (this.state[r][c] !== updatedColumn[r]) {
+            this.state[r][c] = updatedColumn[r];
+            isMovable = true;
+          }
+        }
+      }
+
+      if (isMovable) {
+        this.getRandomCells();
+        this.checkGameStatus();
+      }
+    }
   }
+  moveDown() {
+    if (this.status === Game.Status.playing) {
+      let isMovable = false;
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+      for (let c = 0; c < 4; c++) {
+        const values = [];
+
+        for (let r = 3; r >= 0; r--) {
+          if (this.state[r][c] !== 0) {
+            values.push(this.state[r][c]);
+          }
+        }
+
+        for (let i = 0; i < values.length; i++) {
+          if (values[i] === values[i + 1]) {
+            values[i] *= 2;
+            values[i + 1] = 0;
+            this.score += values[i];
+            isMovable = true;
+          }
+        }
+
+        const updatedColumn = values.filter((value) => value !== 0);
+
+        while (updatedColumn.length < 4) {
+          updatedColumn.push(0);
+        }
+
+        for (let r = 0; r < 4; r++) {
+          if (this.state[r][c] !== updatedColumn[3 - r]) {
+            this.state[r][c] = updatedColumn[3 - r];
+            isMovable = true;
+          }
+        }
+      }
+
+      if (isMovable) {
+        this.getRandomCells();
+        this.checkGameStatus();
+      }
+    }
+  }
+
+  getScore() {
+    return this.score;
+  }
+  getState() {
+    return this.state;
+  }
+  getStatus() {
+    return this.status;
+  }
+
+  start() {
+    this.status = Game.Status.playing;
+    this.getRandomCells();
+    this.getRandomCells();
+  }
+
+  restart() {
+    this.status = Game.Status.idle;
+    this.state = this.initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  getRandomCells() {
+    const emptyCells = [];
+
+    for (let r = 0; r < 4; r++) {
+      for (let c = 0; c < 4; c++) {
+        if (this.state[r][c] === 0) {
+          emptyCells.push([r, c]);
+        }
+      }
+    }
+
+    if (emptyCells.length > 0) {
+      const [randomR, randomC] =
+        emptyCells[Math.floor(Math.random() * emptyCells.length)];
+
+      this.state[randomR][randomC] = Math.random() < 0.9 ? 2 : 4;
+    }
```

Comment:
The method `getRandomCells` modifies the game state directly. It's better to return the new cell's coordinates and value, and let the caller decide how to update the state. This will make your method more flexible and easier to test.